File: Python/student.py
from civitasacademic import CivitasAcademic # Import class CivitasAcademic dari file CivitasAcademic.py
from major import Major # Import class Major dari file major.py

class Student(CivitasAcademic): # Deklarasi class Student (child class dari CivitasAcademic)
    __idNumber = 0 # Untuk menyimpan NIM mahasiswa, atribut berjenis private
    __majorObject = Major("", "", []) # Untuk menyimpan jurusan mahasiswa, atribut berjenis private

    def __init__(self, identity, name, gender, university, email, faculty, idNumber, major): # Membuat constructor
        super().__init__(identity, name, gender, university, email, faculty) # Menginisialisasi setiap atribut yang ada di dalam konstruktor parent class CivitasAcademic
        self.__idNumber = idNumber # Menginisialisasi atribut __idNumber
        self.__majorObject = major # Menginisialisasi atribut __majorObject
    
    def setIdNumber(self, idNumber): # Setter untuk atribut __idNumber
        self.__idNumber = idNumber # Menginisialisasi atribut __idNumber

    def getIdNumber(self): # Getter untuk atribut __idNumber
        return self.__idNumber # Mengembalikan nilai atribut __idNumber
    
    def getMajor(self): # Getter untuk atribut __majorObject
        return self.__majorObject # Mengembalikan nilai atribut __majorObject

File: C++/Student.cpp
// Import library yang dibutuhkan
#include <bits/stdc++.h>

using namespace std; // Menggunakan standard namespace

class Student : public CivitasAcademic // Membuat class Student (child) yang mengextends ke class CivitasAcademic (parent)
{
private:               // Membuat atribut berjenis private
    int idNumber;      // Untuk menyimpan NIM (ID number) mahasiswa
    Major majorObject; // Membuat objek dari class Major

public:
    /* Konstruktor */
    Student() : CivitasAcademic() // Membuat konstruktor tanpa isian, class yang di-extends akan diinstansiasi terlebih dahulu
    {
        this->idNumber = 0;
        this->majorObject = Major();
    }

    Student(long long identity, string name, char gender, string university, string faculty, string email, int idNumber, Major major) : CivitasAcademic(identity, name, gender, university, faculty, email) // Membuat konstruktor dengan isian dari parameter
    {
        this->idNumber = idNumber;
        this->majorObject = major;
    }

    /* Setter dan Getter untuk setiap atribut di dalam class Student */
    void setIdNumber(int idNumber)
    {
        this->idNumber = idNumber;
    }

    int getIdNumber()
    {
        return this->idNumber;
    }

    Major getMajorObject()
    {
        return this->majorObject;
    }

    /* Destruktor */
    ~Student() // Membuat destruktor untuk menghapus semua objek yang telah dibuat
    {
    }
};